### PR TITLE
Reorganize into library

### PR DESCRIPTION
**Pull request by slowli**
This PR changes the type of the cryptocurrency crate to library, and moves the (previously) main binary to examples.

See https://github.com/exonum/cryptocurrency/pull/39#issuecomment-353403775 for context.

### Motivation

Most real-world services are / should be packaged as library crates, so it could be helpful to make the tutorial reasonably close to real-world uses.

### External Changes

If accepted, this PR will require quite a few changes to the service creation tutorial in `exonum-doc`. Nothing to exceptional, though, so I can submit the corresponding changes reasonably fast.

### CODE REVIEW

**Comment by slowli** [comment]
@DenisKolodin @alekseysidorov I made the service schema generic because it's required by business logic-focused testing (#48). I think it could naturally fit in this PR, but I won't insist on it if there are objections.

Also, do you think it's worth the effort to brush over the documentation so it would look adequate if (when?) the crate is published? Or should it be solved separately?

**Comment by slowli** [comment]
@alekseysidorov Re `NodeBuilder`: I think it could deserve a separate PR, quite possibly together with a new tutorial in the overall documentation. I would like to keep the scope of this PR reasonably confined.

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
+Cargo.lock
```
Comment (matklad) [inline .gitignore]: I think if `cryptocurrency` is the main showcase of exonum, which users are supposed to clone and run, then it makes sense to leave `Cargo.lock` in the repository, just to make sure that `cargo build` always goes smoothly (i.e, to protect from a case when some crates does a minor upgrade, which nonetheless has breaking changes).

Comment (slowli) [reply .gitignore]: I see your point. Going a little bit further, a lockfile would be useful in getting the reproducible build of a Docker image with the runnable cryptocurrency example, if we decide to go that way. OTOH, if `cryptocurrency` is intended to be viewed as a legit crate (which was my intention with this PR), then the lockfile is only partially useful; if someone uses it as a crate, it seems to have no use. Is this OK with you?

Comment (matklad) [reply .gitignore]: > if someone uses it as a crate, it seems to have no use. 

This is correct.

>Is this OK with you?

Yep!

**Pull request merged**